### PR TITLE
Finalize login page migration

### LINKED ISSUE #36347: Clean legacy AdminLoginController and related references (links) and templates

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Finalize login page migration
| Type?             | new feature
| Category?         | BO
| BC breaks?        | yes
| Deprecations?     | no
| How to test?      | CI green UI tests green Functional test: the login page in BO, the reset password feature, any campaign/scenario related to login page
| UI Tests          | https://github.com/jolelievre/ga.tests.ui.pr/actions/runs/9676037962 https://github.com/florine2623/testing_pr/actions/runs/9663231021 ✅ 
| Fixed issue or discussion?     | Fixes #36347
| Related PRs       | ~
| Sponsor company   | ~

### CODE REVIEW

**Review by boherm — DISMISSED** [review]

File: install-dev/data/xml/hook.xml
Comment (kpodemski) [inline install-dev/data/xml/hook.xml:889]: @jolelievre do you plan to reintroduce those hooks? one way or another, I'm pretty sure they've been useful for some 2FA modules

Comment (kpodemski) [reply install-dev/data/xml/hook.xml:889]: @jolelievre unless there are certain events triggered on the SF side of things that could be a replacement

**Review by Progi1984 — DISMISSED** [review]
Ok for `tests/UI` directory

File: install-dev/data/xml/hook.xml
Comment (jolelievre) [reply install-dev/data/xml/hook.xml:889]: No these hooks were triggered by the legacy controller, the legacy controller is no more and it wouldn't make sense to maintain them because we can't dispatch the same parameters (which was the legacy controller)

Usually we actually never maintain/remove the hooks removed along with legacy pages, but this time I thought it as worth the extra effort This way I also noted which hooks were actually removed and I'm adding them in the dev doc

My plan is to explain which hook were removed, which ones were maintained (two actually), and what alternative hooks can be used no even if they are different, I'll ping you on the PR for the doc

If we realize some hooks are missing we can also think of what new hooks could/should be introduced

Comment (jolelievre) [reply install-dev/data/xml/hook.xml:889]: There is one hook that was kept strictly for legacy retro compatibility here:
https://github.com/PrestaShop/PrestaShop/blob/29bf2477c2c5fac8b2d51f3dc05d80e46644f88e/src/PrestaShopBundle/Twig/Component/Login/LoginHeadTag.php#L38-L43

It does provide a legacy controller, or actually a "fake" legacy controller we implemented as a bridge to still allow some legacy modifications like `$legacyController->addJs`

Maintaining this hook to allow adding assets was relevant, but all the other hooks would be overkill IMHO It gives the wrong impression that nothing changed and it forces us to maintain retro compatibility in too many places

**Review by matthieu-rolland — DISMISSED** [review]

**Review by florine2623 — COMMENTED** [review]
Hello @jolelievre ,

Tested and compared to old login page.

QA OK ✅ 

- Standard employee email/password login
- Correct employee email/wrong password
- Non-existing email/any password
- Logout from BO
- Links to prestashop-project.org and social medias
- Make sure the work done by @M0rgan01 for backoffice cron calls work correctly with the migrated Login system (search cron regeneration)
- Responsive
- Correct email is sent for Forget my password feature

QA NOK :x: or needs improvement ⚠️ 

## BUG 1
Changing the name of shop (Contact > Stores) doesn't change the title of the login page

Screenshot 2024-06-13 at 14 06 54

## BUG 2
Clickable link "back to FO" is too long. I can click anywhere on the header and it will go back to my FO.
Screenshot 2024-06-13 at 14 00 03

## BUG 3
Non-valid email is not detected. I should have a warning message and I should not be able to click on Log In until my email address has a valid format.

It should be like :
Screenshot 2024-06-13 at 14 04 42

It is like so :
Screenshot 2024-06-13 at 14 04 55

## BUG 4
When a reset link is sent, the message is different between Old and New Login page.
Old Login Page :
Screenshot 2024-06-13 at 14 22 58
New Login page :
Screenshot 2024-06-13 at 14 28 44

## BUG 5
Employees with different roles and permissions cannot access the BO. I created a translator, when I log in with correct email/password, I have an exception. I kept the Default page as Dashboard (not accessible by this employee).
Screenshot 2024-06-13 at 14 29 58

## BUG 6
When my login page has been open long enough and refreshed, I have a CSRF alert. Even after refreshing the page the alert is displayed, I have to manually close the alert.
Screenshot 2024-06-13 at 13 53 55

## BUG 7
Alert message in Log In page is still displayed after I go to Forget my password page. I should not have to manually close it if I move to another page.

https://github.com/PrestaShop/PrestaShop/assets/16019289/9b0fdcf2-f2cc-4056-bcbd-88933b93d959

## BUG 8 
maybe not bug but feature, but I couldn't find any info on that topic)
I can't receive multiple "Forget my password" emails. I have to wait 24 hours. This cannot be configured. In Old Login Page I could send as many as I wanted.
Screenshot 2024-06-13 at 14 51 50

## BUG 9
In Multistore context, I created a second-store, granted access to only 1 employee Translator. Tried to log in via my translator employee account :x: tried to access m store via my default SuperAdmin employee :x: 
BO and login page can't be accessed anymore.
Screenshot 2024-06-13 at 15 01 25

Waiting for your feedback ^^
Thanks!

**Review by boherm — DISMISSED** [review]

**Comment by jolelievre** [comment]
Thanks @florine2623 

here are the bugs I fixed with the recent changes:

**Bug 1** The form title now uses the global configuration (previously it used the Shop->name which stores the shop name in another place)
**Bug 2** Fixed
**Bug 3** The new login page now handles the email validation, the UX is not 100% the same because we use a different BO theme and a different version of bootstrap, but it should be iso-functional besides that.
**Bug 4** The wording was updated to match the previous one (I had copied some other wording for Ajax return, not adapted in this context indeed)
**Bug 5** Actually, the bug was in the employee edition form, when the profile is changed the available options for the home page is supposed to be updated to only propose available values, this feature was broken on develop and is now fixed. If the homepage is edited in some other manual ways (direct DB edit for example) then the access denied is still relevant. However it is handled differently now, we no longer automatically redirect to homepage we display an error page (in prod mode you get a page with the menu so you can navigate to another page).
**Bug 7** Fixed, all the alert messages are removed when we click of `I forgot my password`
**Bug 8** Yes my bad I misunderstood the initial feature, the password reset is only blocked temporarily AFTER you actually reset your password, and the delay is configurable in Advanced Parameters > Team > Password regeneration
**Bug 9** Fixed

This is the only don't that wasn't fixed, but it matches the usual behaviour of Symfony
**Bug 6** This is the native behaviour of the CQRS feature in Symfony, I don't think we should change it, the errors can be closed and on the second try, they should disappear.

**Review by boherm — DISMISSED** [review]

**Review by boherm — DISMISSED** [review]

**Review by matthieu-rolland — DISMISSED** [review]

**Review by boherm — DISMISSED** [review]

**Review by matthieu-rolland — DISMISSED** [review]

**Review by florine2623 — COMMENTED** [review]
Hello @jolelievre ,

Thanks for the fixes !

## QA OK :

BUG 1 ✅ 
BUG 2 ✅ 
## BUG 3 ✅ 
Fixed, email validation is now handled.
Screenshot 2024-06-25 at 15 22 03

## BUG 4 ✅ 
Fixed, message has been updated with correct wording.
Screenshot 2024-06-25 at 15 24 01

BUG 6 ✅ 
BUG 7 ✅ 
BUG 8 ✅ 

## BUG 9 ✅ 
It is fixed. No more exception. My employee has only access to the granted store.
I have found a small bug, I will create an issue. It has nothing to do with the Login page.

Screenshot 2024-06-25 at 16 13 01

## NEED MORE QA : 

## BUG 5 :x:
I see that now, in the Employee edition page, the Default page is changed to an accessible page by the Role of the employee.
But if I change the Default page manually by a page that this employee does not have access to, I still have the Exception.

https://github.com/PrestaShop/PrestaShop/assets/16019289/62b99a0b-f77a-4f3e-88c7-7f090a448102

I think it was better in PS 8.1.x, where I didn't have an exception and I was still able to access my BO with an alert like this :

Screenshot 2024-06-25 at 15 45 58

## BUG 10
(not really a bug, more like an improvement)
Would it be possible to have the same alert in the Password reset page and in Login page when the user does not use a correct email format ?

In **Login page**, I have :
Screenshot 2024-06-25 at 15 22 03

In **Password reset page**, I have :
Screenshot 2024-06-25 at 15 52 54

Thanks!

**Review by boherm — DISMISSED** [review]

**Review by matthieu-rolland — DISMISSED** [review]

**Review by boherm — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Review by florine2623 — APPROVED** [review]
Hello @jolelievre ,

## BUG 5 ✅ 
Works like a charm ! 
The employee has only access to the pages that are accessible by his role !

https://github.com/PrestaShop/PrestaShop/assets/16019289/121ec90e-a781-4e1f-94ba-0c8bf6795de2

## BUG 10 ✅ 
Fixed, same design in login and password page.
Screenshot 2024-06-26 at 11 58 14

Thanks !

It is QA ✅ 💪

**Comment by ps-jarvis** [comment]
QA approved, well done! Message to the maintainers: do not forget to milestone it before the merge.

**Review by kpodemski — APPROVED** [review]

**Comment by kpodemski** [comment]
we need one approval @Progi1984 @nesrineabdmouleh

**Review by Progi1984 — APPROVED** [review]
Ok for UI Tests (`tests/UI/tools/urls.ts`).

Not in my scope for remaining files.

**Comment by kpodemski** [comment]
thanks @jolelievre 🎉

**Comment by gericfo** [comment]
Hello @jolelievre 
Please note that the **“Needs autoupgrade PR”** label has been automatically added to your Pull Request. This means that the changes made in your PR must also be reported in the **Autoupgrade module**, responsible for running the update process of PrestaShop stores. 
When there's a modification to the `db-structure.sql` file or if a Doctrine entity has been modified, added, or deleted, it impacts the database structure. Modifying the list of hooks and feature flags has consequences on the update process as well. These changes can affect how the application interacts with the database and how update procedures are handled. It's important to consider these aspects when contributing to ensure a smooth update experience for users. 
That's why this contribution should be backported, with another corresponding PR, on the **Autoupgrade module's GitHub repository** which is available here: [link](https://github.com/PrestaShop/autoupgrade). 
Don't forget to add a comment in this current thread with the link to this PR once it has been created. 
Thank you.

**Comment by jolelievre** [comment]
Thanks @gericfo 

here is the PR that cleans the removed hooks https://github.com/PrestaShop/autoupgrade/pull/954

**Pull request merged**
